Automattic/wp-calypso
# Purchases: Clarify "Cancel" Copy

Fixes #91760 

## Proposed Changes

* Changes "Cancel" to "Keep plan". 
* Also updates the string to be lower case - I think that all buttons should be (#38231), and I can see that the translation for it already [exists](https://github.com/Automattic/wp-calypso/blob/a900a7934a2be3ef5fe8d78a007e702b820dac38/client/components/marketing-survey/cancel-purchase-form/index.jsx#L646) and is used elsewhere (but let me know if it's intentional!) 

## Why are these changes being made?
See #91760 - the worry is that "Cancel" could be confused as "Cancel [plan]", as opposed to cancelling the action of the dialog. 

## Testing Instructions

Should be just a quick string change for which a code check is sufficient, but you can see this in Calypso by attempting to cancel a plan under Me > Purchases with either WordAds or a custom domain. 

<img width="592" alt="Screenshot 2024-06-26 at 10 15 41" src="https://github.com/Automattic/wp-calypso/assets/43215253/6afaa8a5-fe70-4831-84ff-69c251ed0efa">

cc @michaeldcain
